lib/_emerge/BinpkgExtractorAsync.py

```diff
@@ -129,26 +121,30 @@ def _xpak_start(self):
         self.args = [
             self._shell_binary,
             "-c",
-            (
-                "cmd0=(head -c %d -- %s) cmd1=(%s) cmd2=(tar -xp %s -C %s -f -); "
-                + '"${cmd0[@]}" | "${cmd1[@]}" | "${cmd2[@]}"; '
-                + "p=(${PIPESTATUS[@]}) ; for i in {0..2}; do "
-                + "if [[ ${p[$i]} != 0 && ${p[$i]} != %d ]] ; then "
-                + 'echo command $(eval "echo \\"\'\\${cmd$i[*]}\'\\"") '
-                + "failed with status ${p[$i]} ; exit ${p[$i]} ; fi ; done; "
-                + "if [ ${p[$i]} != 0 ] ; then "
-                + 'echo command $(eval "echo \\"\'\\${cmd$i[*]}\'\\"") '
-                + "failed with status ${p[$i]} ; exit ${p[$i]} ; fi ; "
-                + "exit 0 ;"
+            textwrap.dedent(
+                f"""
+                    cmd0=(head -c {pkg_xpak.filestat.st_size - pkg_xpak.xpaksize} -- {portage._shell_quote(self.pkg_path)})
+                    cmd1=({decomp_cmd})
+                    cmd2=(tar -xp {tar_options} -C {portage._shell_quote(self.image_dir)} -f -);
+                """
+                """
+                    "${cmd0[@]}" | "${cmd1[@]}" | "${cmd2[@]}";
+                    p=(${PIPESTATUS[@]}) ; for i in {0..2}; do
+                """
+                f"""
+                    if [[ ${{p[$i]}} != 0 && ${{p[$i]}} != {128 + signal.SIGPIPE} ]] ; then
+                """
+                """
+                    echo command $(eval "echo \\"'\\${cmd$i[*]}'\\"") failed with status ${p[$i]} ;
+                    exit ${p[$i]} ; fi ; done;
+                    if [ ${p[$i]} != 0 ] ; then
+                    echo command $(eval "echo \\"'\\${cmd$i[*]}'\\"") failed with status ${p[$i]} ;
+                    exit ${p[$i]} ; fi ;
+                    exit 0 ;
+                """
             )
-            % (
-                pkg_xpak.filestat.st_size - pkg_xpak.xpaksize,
-                portage._shell_quote(self.pkg_path),
-                decomp_cmd,
-                tar_options,
-                portage._shell_quote(self.image_dir),
-                128 + signal.SIGPIPE,
-            ),
+            .replace("\n", " ")
+            .strip(),
```

Comment:
@thesamesam @arthurzam What are your opinions on this?

Reply:
I think I can live with the ${{}} and it's more readable this way, but I wouldn't say we should necessarily do it _everywhere_.